src/drivers/uploadthing.ts

```diff
@@ -0,0 +1,90 @@
+import { defineDriver } from "./utils";
+import { ofetch, $Fetch } from "ofetch";
+import { UTApi } from "uploadthing/server";
+
+export interface UploadThingOptions {
+  apiKey: string;
+}
+
+export default defineDriver<UploadThingOptions>((opts) => {
+  let client: UTApi;
+  let utFetch: $Fetch;
+
+  const getClient = () => {
+    return (client ??= new UTApi({
+      apiKey: opts.apiKey,
+      fetch: ofetch.native,
+    }));
+  };
+
+  const getUTFetch = () => {
+    return (utFetch ??= ofetch.create({
+      baseURL: "https://uploadthing.com/api",
+      headers: {
+        "x-uploadthing-api-key": opts.apiKey,
+      },
+    }));
+  };
+
+  return {
+    hasItem(key) {
+      // This is the best endpoint we got currently...
+      return getUTFetch()("/getFileUrl", {
+        body: { fileKeys: [key] },
+      }).then((res) => res.ok);
+    },
+    getItem(key) {
+      return ofetch(`https://utfs.io/f/${key}`);
```

Comment:
Maybe, somehow make it configurable too? (not sure about `/api` and this API difference but just a taught to again make testing easier)

Reply:
utfs is our cloudflare rewriter that handles caching and mapping keys to different regions.

uploadthing.com/api is where everything else goes.

can make it configurable for testing though!